library/src/main/java/com/github/mproberts/rxdatabinding/notifications/NotificationBindingHandler.java

```diff
@@ -319,7 +321,7 @@ public void setList(FlowableList<T> list) {
         _boundList = list;
 
         if (list != null) {
-            Disposable subscription = list.updates().subscribe(new Consumer<Update<T>>() {
+            _listSubscription = list.updates().subscribeOn(UiThreadScheduler.uiThread()).subscribe(new Consumer<Update<T>>() {
```

Comment:
This should be an `observeOn` call not `subscribeOn`, since it's flowing down the chain to the subscribe callback.